lib/meow/report.ex

```diff
@@ -210,10 +215,105 @@ defmodule Meow.Report do
         %{population: "population #{idx}", generation: population.generation}
       end
 
-    Vl.new(width: 300)
+    Vl.new(title: "Generations", width: 300)
     |> Vl.data_from_values(data)
     |> Vl.mark(:bar)
     |> Vl.encode_field(:x, "generation", type: :quantitative)
     |> Vl.encode_field(:y, "population", type: :nominal, axis: [title: nil])
   end
+
+  @doc """
+  Saves an HTML report including the summary and plots to a specified path.
+  """
+  @spec to_html(t(), Path.t()) :: :ok | {:error, File.posix()}
+  def to_html(report, path) do
+    assert_vega_lite!("to_html/2")
+
+    summary = format_summary(report)
+
+    plots = metrics_plots(report) ++ [plot_times(report)] ++ [plot_generations(report)]
+
+    plot_divs =
+      1..length(plots)
+      |> Enum.map(fn i -> ~s{<div id="plot-#{i}" class="plot"></div>} end)
+      |> Enum.join("\n")
+
+    script = plots_script(plots)
+
+    html = """
+    <!DOCTYPE html>
+    <html lang="en">
+    <head>
+      <meta charset="UTF-8">
+      <meta name="viewport" content="width=device-width, initial-scale=1.0">
+      <title>Meow Report</title>
+      <script src="https://cdn.jsdelivr.net/npm/vega@5.20.2"></script>
+      <script src="https://cdn.jsdelivr.net/npm/vega-lite@5.1.0"></script>
+      <script src="https://cdn.jsdelivr.net/npm/vega-embed@6.17.0"></script>
+
+      <style>
+        .container {
+          display: flex;
+          flex-direction: column;
+          align-items: center;
+          max-width: 900px;
+          margin: 0 auto;
+          padding: 1rem;
+        }
+
+        .header {
+          width: 100%;
+        }
+
+        .plot:not(:first-of-type) {
+          margin-top: 32px;
```

Comment:
Let's do even more, like `64px` or so. Also, it would actually make sense to have the margin between all container elements, not just plots.